simulator/src/muon.rs

```diff
@@ -0,0 +1,210 @@
+use super::json;
+use supermusr_common::{Intensity, Time};
+
+#[derive(Debug)]
+pub(crate) enum Muon {
```

Comment:
This name is a tad ambiguous, I assume this is modelling a muon decay?

Reply:
This models a muon detection event.

Reply:
In that case could this be renamed to something that conveys that.